src/synthesizer/blackholes.py

```diff
@@ -0,0 +1,460 @@
+"""
+A generic blackholes class currently holding (ultimately) various blackhole 
+emission models. 
+
+TODO: this should probably be moved into components/blackholes.py
+"""
+
+import numpy as np
+from unyt import deg, km, cm, s, K, rad
+from synthesizer.dust.emission import Greybody
+from synthesizer.grid import Grid
+from synthesizer.sed import Sed
+from synthesizer.exceptions import MissingArgument
+
+
+class BlackHoleEmissionModel:
+    """
+    Potential parent class for blackhole emission models.
+    """
+
+    pass
+
+
+class Template(BlackHoleEmissionModel):
+
+    """
+    Use a template for the emission model. 
+
+    The template is simply scaled by bolometric luminosity.
+    
+    """
+
+    def __init__(self, filename=None, lam=None, lnu=None):
+
+        """
+        
+        Arguments:
+            filename (str)
+                The filename (including full path) to a file containing the 
+                template. The file should contain two columns with wavelength
+                and luminosity (lnu).
+            lam (array)
+                Wavelength array.
+            lnu (array)
+                Luminosity array.
+        
+        """
+
+        if filename:
+
+            # to be implemented implement later
```

Comment:
And an `UnimplementedFunctionality` error

Reply:
Done